apps/settings/js/modules/base/event_emitter.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * EventEmitter implements basic event emitting logics. It also provides
+ * function that augments the ability to other objects. For example:
+ *  
+ *   function Module() {
+ *     this.EventEmitter.call(this, [eventName1, eventName2]);
```

Comment:
This line is considered as calling to the constructor of `EventEmitter`. If the constructor is not called correctly, the emitter won't work. Just like prototypical inheritance.